app/models/bundle_context.rb

```diff
@@ -27,6 +24,11 @@ class BundleContext < ApplicationRecord
     "smpl_cm_style" => 2
   }
 
+  # return [PreAssembly::Bundle]
+  def bundle
+    @bundle ||= PreAssembly::Bundle.new(self)
+  end
```

Comment:
I'd have to look at the whiteboard by Jeremy's desk to refresh my memory:  is it possible for a bundle_context to have more than one bundle?   I know a bundle can only have a single bundle_context ...

Reply:
Keeping in mind that `Bundle` is not separately persisted, I think there only ever a need for one Bundle in memory at a given time for a given BC.

Reply:
Hmmm.  I think what I'm remembering from the whiteboard is that JobRuns pertain to bundles, which have bundle contexts.  So perhaps the right thing to pass to a JobRun is the bundle, not the bundle_context - you can already get the bundle_context from the bundle/

Reply:
this method seems like a reasonable thing to have:  a `Bundle` is instantiated solely from a `BundleContext`, used to run pre-assembly or discovery report, and left unpersisted.  so this seems like a reasonable convenience method for obtaining a usable `Bundle` instance when given a `BundleContext`.

i would agree that some specific commentary about where this method is intended to be used would be nice, if only in the PR description or comments.  i'm guessing something like, "job gets instantiated with a `JobRun` instance, uses the `JobRun` to get the `BundleContext`, which it uses to instantiate `Bundle` and do some work".

there are actually only two instances of `Bundle.new` in master right now, both in spec code.  per F2F conversation w/ @ndushay this morning, i'd be inclined to file a small touch-up ticket for switching that code to use this approach if it gets merged as-is (which is what i'm guessing will happen).

Reply:
I don't think anything here *requires* that you initialize via this method, but `PreAssembly::Bundle#initialize` already requires a `BundleContext`, so you will already have a BC somewhere beforehand, so might as well use this method.

Reply:
per F2F discussion, projected for use as a convenience method by `*Job` (which has a `JobRun`, which has a `BundleContext`).